lib/splitclient-rb/split_factory.rb

```diff
@@ -17,12 +17,18 @@ def initialize(api_key, config_hash = {})
       @metrics_repository = MetricsRepository.new(@config.metrics_adapter, @config)
 
       @sdk_blocker = SDKBlocker.new(@config, @splits_repository, @segments_repository)
-      @adapter = SplitAdapter.new(@api_key, @config, @splits_repository, @segments_repository, @impressions_repository, @metrics_repository, @sdk_blocker)
+      @adapter = start!
```

Comment:
@patricioe  In Ruby we add bang to the end of the method name when it is "dangerous". Often it means that it changes the state, e.g. we have 2 methods `map` and `map!`. First one will return a modified copy, whereas the second one will modify the object itself. By adding bang to this method I want to show that this method changes the state as well - it changes cache. This 100% not required, but considered good coding style